### DISCUSSION TITLE
support margin setting in image directive syntax

### DISCUSSION THREAD

**Discussion by chenxizhang** [marp-team/marp#283] [Ideas] [answered]
it's better to support margin setting in image directive syntax.

![image](https://user-images.githubusercontent.com/1996954/158809940-46e7a480-43f5-4670-9cd3-cee4a02236c8.png)

For example

`![bg fit left margin:10px](images/excel.png)`

**Comment by chenxizhang** [comment]
I have a workaround on this. I created a css file including all the customized style, 
![image](https://user-images.githubusercontent.com/1996954/159102401-45676c5d-e897-42fe-8711-a43ce0723f76.png)

and put it into the output html thought the custom engine
![image](https://user-images.githubusercontent.com/1996954/159102434-4fcd65ae-aa3c-4098-be94-3e5d5bba5587.png)

it works for me. just a little unhappy thing is I can't see the effect during the preview mode, although I can see it after build it. any idea?

**Reply by chenxizhang** [reply]
Now, it's better

![image](https://user-images.githubusercontent.com/1996954/159102736-1f23f7fb-a4c6-4c98-be3b-de85e9e95d6a.png)

**Comment by chenxizhang** [comment]
I checked the document many times, is there any easy way to define a custom css (or theme)? I can't find anything about how to do this.

https://marpit.marp.app/theme-css

**Comment by chenxizhang** [comment]
I found a way to do this in preview mode

![image](https://user-images.githubusercontent.com/1996954/159103464-074c2564-7f84-4689-84f9-d8173d1938ce.png)

**Comment by chenxizhang** [comment]
```css
/* @theme xizhang */

@import 'default';

section footer{
    margin-left: 50px !important;
}
/* 这个设置适合于所有的背景图片*/
section figure{
    margin-right: 10px !important;
}

/* 这个设置适合于左侧的背景图片 */
section figure:first-child{
    margin-left: 10px !important;
}

```

**Comment by yhatt** [comment, marked as answer]
We have no plan to implement margin syntax. The image syntax by Marpit framework is based on ability of `background` CSS property, and there is no ability to set margin for background image in [the bunch of CSS background properties](https://developer.mozilla.org/en-US/docs/Web/CSS/background#constituent_properties). 

For Marp users, setting the margin to `figure` element through `<style scoped>` is a good workaround. On the other hand, the margin syntax would make inequality between the simple background mode (based only background CSS properties) and advanced background mode (using `figure` element) if added to Marpit framework.

If you want to make a margin around of background area by using exist image syntax, just scaling down background image to under `100%` would get better result.

```markdown
![bg 80%](https://example.com/image.jpg)
```

**Reply by chenxizhang** [reply]
Thanks for the reply. Understood for the consideration. And currently my workaround is just work for me.

I will try your suggestion to see if it can help although I often use `fit` with the `bg` syntax.

**Comment by kongdd** [comment]
I need to decrease the margin of the left section. Do you have any suggestion?

![image](https://user-images.githubusercontent.com/9815742/201101772-a3a92955-9379-42ce-92a1-251dbbc02cbc.png)

```md
<style scoped>
/*这个设置适合于所有的背景图片*/
section {
   margin: 0em 0em 0.1em 0em
   padding: 0px;
}
</style>

1. 水汽充足情况下：
   X -> A点

2. 能量平衡情况下：
   X -> Y点

![bg right:70% 90%](images/感热与潜热比例.png)

```

**Reply by yhatt** [reply]
Your style is not recognized due to missing semicolon at the last of `margin` declaration.

```diff
 <style scoped>
 /*这个设置适合于所有的背景图片*/
 section {
-   margin: 0em 0em 0.1em 0em
+   margin: 0em 0em 0.1em 0em;
    padding: 0px;
 }
 </style>
 
 1. 水汽充足情况下：
    X -> A点
 
 2. 能量平衡情况下：
    X -> Y点
 
 ![bg right:70% 90%](images/感热与潜热比例.png)
```

Example in regular HTML document + CSS: https://jsfiddle.net/84q97fjv/

**Reply by kongdd** [reply]
Thanks very much for your quick reply.
This is my another solution:
```css
section[data-marpit-advanced-background] {
  padding: 5px;
}
```

**Comment by chenxizhang** [comment]
I used obsidian for a while, and it also supports the presenter mode, but it is very simple and I think the great work you have done, can definitely help them to improve.
